bwapi/bwresources.py

```diff
@@ -114,12 +156,13 @@ def upload_all(self, data_list, create_only=False, modify_only=False):
             filled_data = self._fill_data(data)
             name = data["name"]
 
-            if name in self.ids and not create_only:
+            if self.get_resource_id(name,check=True) and not create_only: #if resource does exist
```

Comment:
creating an `if exists` method also negates the need for this explanatory comment 😉

Reply:
true - created this method (check_resource_exists())